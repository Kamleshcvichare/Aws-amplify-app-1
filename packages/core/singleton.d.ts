export * from './lib-esm/Singleton';