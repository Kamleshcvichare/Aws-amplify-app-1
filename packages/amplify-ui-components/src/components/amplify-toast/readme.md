# amplify-toast



<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                                                            | Type         | Default     |
| --------- | --------- | ---------------------------------------------------------------------- | ------------ | ----------- |
| `onClose` | --        | Used in order to add a clickable `exit` button for the Toast component | `() => void` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
