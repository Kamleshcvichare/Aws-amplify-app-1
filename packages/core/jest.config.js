module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
