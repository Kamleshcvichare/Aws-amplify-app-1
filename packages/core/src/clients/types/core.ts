/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
