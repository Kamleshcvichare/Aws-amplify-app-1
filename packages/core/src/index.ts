// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
This file maps top-level exports from `@aws-amplify/core`. These are intended to be potentially customer-facing exports.
*/
// Hub exports
export { Hub } from './Hub';
export { HubCapsule, HubCallback, HubPayload } from './Hub/types';

// Singleton exports
export {
	TokenProvider,
	AuthTokens,
	FetchAuthSessionOptions,
	AWSCredentialsAndIdentityIdProvider,
	AWSCredentialsAndIdentityId,
	Identity,
	OAuthConfig,
	CognitoUserPoolConfig,
} from './singleton/Auth/types';
export {
	AuthConfig,
	AuthUserPoolConfig,
	AuthUserPoolAndIdentityPoolConfig,
	APIConfig,
	CacheConfig,
	StorageAccessLevel,
	StorageConfig,
	GetCredentialsOptions,
	ResourcesConfig,
	LibraryOptions,
	AnalyticsConfig,
	GeoConfig,
} from './singleton/types';
export {
	Amplify,
	fetchAuthSession,
	AmplifyClass as AmplifyClassV6,
	clearCredentials,
} from './singleton';

// AWSClients exports
export {
	getCredentialsForIdentity,
	getId,
	GetCredentialsForIdentityInput,
	GetCredentialsForIdentityOutput,
} from './awsClients/cognitoIdentity';

// Amplify-wide constructs
export { UserProfile } from './types';

// Storage helpers
export {
	CookieStorage,
	defaultStorage,
	sessionStorage,
	sharedInMemoryStorage,
} from './storage';
export { KeyValueStorageInterface } from './types';

// Cache exports
export { Cache } from './Cache';

// Internationalization utilities
export { I18n } from './I18n';

// Logging utilities
export { ConsoleLogger } from './Logger';
