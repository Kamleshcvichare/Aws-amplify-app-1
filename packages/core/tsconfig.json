{
	"extends": "../../tsconfig.json",
	"compilerOptions": {
		"baseUrl": ".",
		"importHelpers": false,
		"paths": {
			"~/*": ["./*"]
		}
	},
	"include": ["./src", "__tests__"]
}
