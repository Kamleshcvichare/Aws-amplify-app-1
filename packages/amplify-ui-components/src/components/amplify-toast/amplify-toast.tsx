import { Component, Prop, h } from '@stencil/core';

import { toast, toastClose } from './amplify-toast.style';

@Component({
  tag: 'amplify-toast',
  shadow: false,
})
export class AmplifyToast {
  /** Used in order to add a clickable `exit` button for the Toast component */
  @Prop() onClose: () => void;

  render() {
    return (
      <div class={toast}>
        <slot />
        <a class={toastClose} onClick={this.onClose} />
      </div>
    );
  }
}
