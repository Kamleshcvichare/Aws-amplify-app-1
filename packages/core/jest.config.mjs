import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
