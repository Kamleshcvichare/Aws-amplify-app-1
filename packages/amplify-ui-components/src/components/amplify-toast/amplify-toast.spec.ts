import { newSpecPage } from '@stencil/core/testing';
import { AmplifyToast } from './amplify-toast';

describe('amplify-toast spec:', () => {
  describe('Component logic ->', () => {
    let toast;

    beforeEach(() => {
      toast = new AmplifyToast();
    });

    it('should have `onClose` be undefined by default', () => {
      expect(toast.onClose).toBeUndefined();
    });
  });
  describe('Render logic ->', () => {
    it('should render `Toast!`', async () => {
      const page = await newSpecPage({
        components: [AmplifyToast],
        html: `<amplify-toast></amplify-toast>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
