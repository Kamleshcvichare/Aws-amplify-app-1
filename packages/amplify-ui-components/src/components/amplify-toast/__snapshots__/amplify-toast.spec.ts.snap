// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-toast spec: Render logic -> should render \`Toast!\` 1`] = `
<amplify-toast>
  <!---->
  <div class="css-1toxen">
    <a class="css-1s6iclz"></a>
  </div>
</amplify-toast>
`;
